src/state/notifications/sagas.ts

```diff
@@ -72,9 +73,18 @@ export function* verifyNotificationEmailSaga(action: VerifyNotificationEmailActi
 }
 
 export function* subscribeWalletSaga(action: SubscribeWalletAction) {
-  const { payload } = action as SubscribeWalletAction;
+  const {
+    payload: { wallets, email },
+  } = action as SubscribeWalletAction;
   try {
-    const response: { session_token: string; result: Result } = yield call(subscribeEmail, payload);
+    const walletsGenerationBase = yield all(wallets.map(wallet => call(walletToAddressesGenerationBase, wallet)));
+
+    const lang = yield select(appSettingsSelectors.language);
+    const response: { session_token: string; result: Result } = yield call(subscribeEmail, {
+      email,
+      wallets: walletsGenerationBase,
+      lang,
+    });
     if (response.session_token) {
       yield put(subscribeWalletSuccess(response.session_token));
```

Comment:
we dont handle results.failure anyhow? No error message, no info, no throw, no nothing?

Reply:
This question apply to all API calls

Reply:
we handle it, subscribe email throws an error that is caught by try-catch